src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class FileContentResultExecutor : FileResultExecutor
+    {
+        private readonly ILogger _logger;
+
+        public FileContentResultExecutor(ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _logger = loggerFactory.CreateLogger<FileContentResultExecutor>();
+        }
+
+        public Task ExecuteAsync(ActionContext context, FileContentResult result)
+        {
+            SetHeaders(context, result);
+            _logger.FileResultExecuting(result.FileDownloadName);
+
+            return WriteFileAsync(context, result);
+        }
+
+        private Task WriteFileAsync(ActionContext context, FileContentResult result)
+        {
+            var response = context.HttpContext.Response;
+
+            var bufferingFeature = response.HttpContext.Features.Get<IHttpBufferingFeature>();
+            bufferingFeature?.DisableResponseBuffering();
```

Comment:
@Tratcher We do it in a bunch of places. Is there any downside to it if its not implemented? Will we get surprised by it in a negative way if Kestrel or any other of our components decides to implement it?